tests/unit/array/mixins/test_find.py

```diff
@@ -987,3 +987,116 @@ class MMDoc:
     assert (closest_docs[0].embedding == np.array([3, 3])).all()
     for d in closest_docs:
         assert d.id.endswith('_2')
+
+
+@pytest.mark.parametrize(
+    'storage, config, subindex_configs',
+    [
+        ('memory', None, {'@c': None}),
+        (
+            'weaviate',
+            {
+                'n_dim': 3,
+            },
+            {'@c': {'n_dim': 3}},
+        ),
+        ('annlite', {'n_dim': 3}, {'@c': {'n_dim': 3}}),
+        ('sqlite', dict(), {'@c': dict()}),
+        ('qdrant', {'n_dim': 3}, {'@c': {'n_dim': 3}}),
+        ('elasticsearch', {'n_dim': 3}, {'@c': {'n_dim': 3}}),
+        ('redis', {'n_dim': 3}, {'@c': {'n_dim': 3}}),
+        ('milvus', {'n_dim': 3}, {'@c': {'n_dim': 3}}),
+    ],
+)
+def test_find_return_root(storage, config, subindex_configs, start_storage):
```

Comment:
I would like to see two more test cases (can be in the same test): That the returned root-level documents are actually in the right order, and that the associated scores are copied correctly.